youtube_dl/extractor/pandatv.py

```diff
@@ -0,0 +1,78 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..compat import compat_str
+from ..utils import (
+    ExtractorError,
+    qualities
+)
+
+class PandaTVIE(InfoExtractor):
+    IE_DESC = '熊猫TV'
+    _VALID_URL = r'http://(?:www\.)?panda\.tv/(?P<id>[0-9]+)'
+    _TESTS = [{
+        'url': 'http://www.panda.tv/10091',
+        'info_dict': {
+            'id': '10091',
+            'title': 're:.+',
+            'uploader': '囚徒',
+            'ext': 'flv',
+            'is_live': True,
+        },
+        'params': {
+            'skip_download': True,
+        },
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        config = self._download_json(
+            'http://www.panda.tv/api_room?roomid=%s' % video_id,
+            video_id
+        )
+
+        data = config['data']
+
+        error_code = config.get('errno', 0)
+        if error_code is not 0:
+            error_desc = 'Server reported error %i' % error_code
+            if isinstance(data, compat_str):
+                error_desc += ': ' + data
+            raise ExtractorError(error_desc, expected=True)
+
+        video_info = data.get('videoinfo')
+
+        # 2 = live, 3 = offline
+        if video_info.get('status') != '2':
+            raise ExtractorError(
+                'Live stream is offline', expected=True)
+
+        title = data['roominfo']['name']
+        uploader = data.get('hostinfo', {}).get('name')
+        plflag = video_info['plflag'].split('_')[1]
+        room_key = video_info['room_key']
+        stream_addr = video_info['stream_addr']
+
+        quality_key = qualities(['HD', 'SD'])
+        suffix = ['_mid', '']
```

Comment:
Use browser to watch the streams, if both `高清(720P)` and `超清(1080P)` are selectable in the flash player(e.g. http://www.panda.tv/371037), `stream_addr` is:

    "stream_addr": {"OD": "0", "HD": "1", "SD": "1"}

If `超清(1080P)` is the only option(e.g. http://www.panda.tv/359501), `stream_addr` is:

    "stream_addr": {"OD": "0", "HD": "0", "SD": "1"}

My guess is :

    SD -> Super Definition(not Standard Definition) -> 超(means super)清(1080P) -> ''
    HD -> High Definition -> 高(means high)清(720P) -> _mid
    OD -> Ordinary Definition -> 标清(540P) -> _small

That's how most video web sites in China define `高清(720P)` and `超清(1080P)`。

The `_small` quality is never available in the flash player, I've only seen it in the pandatv mobile app. Maybe the mobile app use a different api to get the info, rather than `http://www.panda.tv/api_room?roomid=<roomid>`.